### PR TITLE
Fixes Exceptions on Rejoining

### PR DESCRIPTION
**Pull request by clusterfack**
This fixes two different exceptions you received when rejoining

1. A server exception that occurred due to the fact that the gamestatemanager would get cleared if everyone disconnected, but the list that it uses as keys to itself would not get cleared. Now both are cleared.

2. A client exception that occurred due to the component references list being completely cleared. This doesn't get regenerated upon reconnect, @PJB3005 may have to say whether or not this is wise but you cannot reconnect with it there.

Client still looks like shit when you reconnect though, and you cant move your player character, and the console has a line that says player.name variable is already defined.

### CODE REVIEW

File: SS14.Shared/GameObjects/Entity.cs [hunk 1]
```diff
@@ -421,7 +421,7 @@ public virtual void Shutdown()
             }
             _components.Clear();
             _netIDs.Clear();
-            _componentReferences.Clear();
+//          _componentReferences.Clear();
```
Comment (PJB3005) [inline SS14.Shared/GameObjects/Entity.cs]: Wait what? Are these entities being re-used after you reconnect?

Comment (clusterfack) [inline SS14.Shared/GameObjects/Entity.cs]: It seems so, if a client reconnects after this list gets cleared it getcomponent which uses this list immediately has an exception from trying to read it.

Comment (PJB3005) [reply SS14.Shared/GameObjects/Entity.cs]: Ok these entities shouldn't be getting reused. Fix that instead of "solving" the exception without fixing it.

**Comment by clusterfack** [comment]
Alright, I implemented PJB's suggestion and it seems to have fully brought the functionality back to what it used to be before.

**Pull request merged**